app/src/main/java/org/sagebionetworks/research/mpower/research/MpIdentifier.java

```diff
@@ -0,0 +1,31 @@
+package org.sagebionetworks.research.mpower.research;
+
+import android.support.annotation.StringDef;
+
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.SOURCE)
+@StringDef({ MpIdentifier.TRIGGERS, MpIdentifier.SYMPTOMS, MpIdentifier.MEDICATION,
+        MpIdentifier.STUDY_BURST_COMPLETED, MpIdentifier.TAPPING, MpIdentifier.WALK_AND_BALANCE,
+        MpIdentifier.TREMOR, MpIdentifier.DEMOGRAPHICS, MpIdentifier.BACKGROUND,
+        MpIdentifier.ENGAGEMENT, MpIdentifier.MOTIVATION, MpIdentifier.STUDY_BURST_REMINDER,
+        MpIdentifier.MEASURING, MpIdentifier.TRACKING, MpIdentifier.HEALTH_SURVEYS })
+public @interface MpIdentifier {
```

Comment:
Should probably replace/merge this with org.sagebionetworks.research.mpower .Task

Reply:
I've been having trouble finding classes in AS lately that are mentioned I wonder what's going on with the class search function.  I can't find "Task" with identifiers, or the "TaskInfo" object you mentioned previously.

Reply:
> > Slight quibble with Michael's very good description. We could have TodayViewController and StudyBurstViewController share an instance of the StudyBurstViewModel by passing the same lifecycle object. You'd pick whichever Fragment has the longer lifecycle, or if they are the same, a shared ancestor Fragment or Activity. Nice way to communicate or share state between Fragments compared to the old way of passing messages.
> > Also, ViewModels do often contain state.
> 
> Sure, we could dispatch the ScheduleViewModel that way, but what problem would that be solving? Everything in that class is reactive to LiveData links to the Room Schedule DB which keeps it synced no matter who owns it or how many instances there are.

We don't need to make this change for mPower. More generally, doing it this way supports the features needed in UX in other apps. Shannon and I talked more about how that would be supported.

Yes, reactive streams are pretty great :)